File: MoveZeroes/move_zeroes.go
func moveZeroes(nums []int)  {
    i, j, n := 0, 0, len(nums)
    for j < n {
        nums[i] = nums[j]
        j++
        if nums[i] != 0 {
            i++
        }
    }
    for i < n {
        nums[i] = 0
        i++
    }
}

File: MoveZeroes/move_zeroes.c
void moveZeroes(int* nums, int numsSize)
{
	    int i = 0, j = 0;
	    while (j < numsSize) {
		        nums[i] = nums[j++];
		        i += nums[i] != 0;
	    }
	    memset(nums + i, 0, (numsSize - i) * sizeof(int));
}